signac/core/jsondict.py

```diff
@@ -194,8 +194,41 @@ def buffer_reads_writes(buffer_size=DEFAULT_BUFFER_SIZE, force_write=False):
 
 
 class JSONDict(SyncedAttrDict):
-
-    def __init__(self, parent=None, filename=None, write_concern=False):
+    """A dict-like mapping interface to a persistent JSON-file.
+
+    The JSONDict is a :class:`~collections.abc.MutableMapping` and therefore
+    behaves similar to a :class:`dict`, but all data is stored persistently in
+    the associated JSON-file on disk.
+
+    .. code-block:: python
+
+            doc = JSONDict('data.json', write_concern=True)
```

Comment:
Is this indented too far? Maybe 4 instead of 8? (Not sure what the standard is across the code.)